multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineVcapServicesPropertiesChangedStep.java

```diff
@@ -0,0 +1,44 @@
+package org.cloudfoundry.multiapps.controller.process.steps;
+
+import java.text.MessageFormat;
+import java.util.List;
+
+import javax.inject.Named;
+
+import org.cloudfoundry.multiapps.controller.client.lib.domain.CloudApplicationExtended;
+import org.cloudfoundry.multiapps.controller.process.Messages;
+import org.cloudfoundry.multiapps.controller.process.listeners.BindUnbindServiceEndListener;
+import org.cloudfoundry.multiapps.controller.process.variables.Variables;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+@Named("determineVcapServicesPropertiesChangedStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class DetermineVcapServicesPropertiesChangedStep extends SyncFlowableStep {
+
+    @Override
+    protected StepPhase executeStep(ProcessContext context) {
+        List<String> services = context.getVariable(Variables.SERVICES_TO_UNBIND_BIND);
+        CloudApplicationExtended app = context.getVariable(Variables.APP_TO_PROCESS);
+
+        boolean changedVcapServicesProperties = services.stream()
+                                                        .map(service -> BindUnbindServiceEndListener.buildExportedVariableName(app.getName(),
+                                                                                                                               service))
+                                                        .map(variableName -> StepsUtil.getObject(context.getExecution(), variableName))
```

Comment:
`services.stream()
                                                        .map(service -> BindUnbindServiceEndListener.buildExportedVariableName(app.getName(),
                                                                                                                               service))
                                                        .map(variableName -> StepsUtil.getObject(context.getExecution(), variableName))
                                                        .filter(Boolean.class::isInstance)
                                                        .anyMatch(Boolean.class::cast);`

Reply:
Can be simplified a bit.
Is this filter needed: filter(Boolean.class::isInstance). It should always be boolean?
Can you extract it in a separate method?